features/FxA-45-device-registration-api/README.md

```diff
@@ -0,0 +1,360 @@
+
+#### Response
+
+```
+{
+  "uid": "4c352927cd4f4a4aa03d7d1893d950b8",
+  "sessionToken": "27cd4f4a4aa03d7d186a2ec81cbf19d5c8a604713362df9ee15c4f4a4aa03d7d",
+  "keyFetchToken": "7d1893d950b8cd69856a2ec81cbfd7d1893d950b3362df9e56a2ec81cbf19d5c",
+  "authAt": 1392144866,
+  "device": {
+    "id": "3d7d",
+    "name": "My Phone",
+    "type": "mobile",
+    "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+  }
+}
+```
+
+### POST /v1/account/device
+
+Authenticated with session token or OAuth.
+
+#### Requests
+
+##### New device
+
+```
+curl -v \
+-X POST \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/device \
+-d '{
+  "name": "My Phone",
+  "type": "mobile",
+  "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+}'
+```
+
+##### Registered device
+
+```
+curl -v \
+-X POST \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/device \
+-d '{
+  "id": "3d7d",
+  "name": "My Old Phone"
+}'
+```
+
+#### Response
+
+```
+{
+  "id": "3d7d",
+  "name": "My Phone",
+  "type": "mobile",
+  "callback": "https://updates.push.services.mozilla.com/update/abcdef01234567890abcdefabcdef01234567890abcdef"
+}
+```
+
+### POST /v1/account/device/destroy
+
+Authenticated with session token or OAuth.
+
+#### Request
+
+```
+curl -v \
+-X POST \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/device/destroy \
+-d '{
+  "id": "3d7d"
+}'
+```
+
+#### Response
+
+```
+{}
+```
+
+### GET /v1/account/devices
+
+Authenticated with session token or OAuth.
+
+#### Request
+
+```
+curl -v \
+-X GET \
+-H "Host: api-accounts.dev.lcip.org" \
+-H "Content-Type: application/json" \
+-H 'Authorization: Hawk id="d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c7509c5632ac35b28b48d", ts="1373391043", nonce="ohQjqb", hash="vBODPWhDhiRWM4tmI9qp+np+3aoqEFzdGuGk0h7bh9w=", mac="LAnpP3P2PXelC6hUoUaHP72nCqY5Iibaa3eeiGBqIIU="' \
+https://api-accounts.dev.lcip.org/v1/account/devices
+```
+
+#### Response
+
+Instead of including the `sessionToken` field,
```

Comment:
@shane-tomlinson we may actually need to return the sessionToken here, depending on the logic that will be used by the devices view.  If it wants to display things that are sessions but don't have a device record, then it will need to be able to match each device record back to its current session token.